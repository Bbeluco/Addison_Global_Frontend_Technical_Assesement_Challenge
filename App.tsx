import 'react-native-gesture-handler';
import React, { useEffect } from 'react';
import BetSlip from './src/components/betslip/BetSlip';
import MktpBet from './src/components/mktpBet/MktpBet';
import { createDrawerNavigator } from '@react-navigation/drawer';
import { NavigationContainer } from '@react-navigation/native';
import useMktpBetHooks from './src/components/mktpBet/MktpBetHooks';
import { getBetsAvailable } from './src/requests/apiRequestInfo';
import { ScrollView } from 'react-native-gesture-handler';

export default function App() {
  const { 
    betOption, 
    pushSelectedOptionInfo, 
    removeItemFromArray,
    apiResponseBetAvaible, 
    setApiResponseBetAvaible } = useMktpBetHooks()
  const Drawer = createDrawerNavigator();

  useEffect(() => {
    const response = async () => {
      return await getBetsAvailable()
    }
    response()
      .then((res) => {
          setApiResponseBetAvaible(res.data)
      }).catch(e => {
        console.log(e)
      })
  }, [])


  return (
    <NavigationContainer>
        <Drawer.Navigator 
          useLegacyImplementation={false}
          drawerContent={({ state, navigation}) => <BetSlip betOptions={betOption} removeItemFromArray={removeItemFromArray} closeDrawer={navigation.closeDrawer}/>}>
          <Drawer.Screen name='Bets'>
            {() => <MktpBet 
                      betOption={betOption} 
                      pushSelectedOptionInfo={pushSelectedOptionInfo} 
                      apiResponseBetAvaible={apiResponseBetAvaible}
                    />}
          </Drawer.Screen>
        </Drawer.Navigator>
    </NavigationContainer>
  );
}