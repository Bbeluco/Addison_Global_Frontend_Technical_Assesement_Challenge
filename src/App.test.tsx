import App from "../App";
import axios from "axios";
import renderer, { act } from 'react-test-renderer';
import mockReturnApi from "./mocks/mockReturnApi";
import { fireEvent, render, screen } from "@testing-library/react-native";
import { waitFor } from "@testing-library/dom";

jest.mock('axios')

beforeAll(() => {
    (axios.get as jest.Mock).mockResolvedValue({ data: mockReturnApi});
})

describe('<App />', () => {
    it('should render UI correctly', async () => {
        const component = renderer.create(<App/>).toJSON();
        await waitFor(() => {
            screen.findByText(mockReturnApi[0]['markets'][0]['name'])
        })
        expect(component).toMatchSnapshot();
    })

    it('should interact with any option available', async () => {
        render(<App />)
        await waitFor(() => {
            screen.findByText(mockReturnApi[0]['markets'][0]['name'])
        })

        screen.debug()
    })
})