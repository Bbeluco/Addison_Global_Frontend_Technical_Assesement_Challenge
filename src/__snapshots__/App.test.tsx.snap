// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render UI correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      collapsable={false}
      handlerTag={1}
      handlerType="PanGestureHandler"
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      style={
        [
          {
            "flex": 1,
            "overflow": "hidden",
          },
          {
            "flexDirection": "row",
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "flex": 1,
            },
            {
              "transform": [
                {
                  "translateX": 0,
                },
              ],
            },
          ]
        }
      >
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          style={
            {
              "flex": 1,
            }
          }
        >
          <RNSScreenNavigationContainer
            style={
              {
                "flex": 1,
              }
            }
          >
            <RNSScreen
              activityState={2}
              collapsable={false}
              gestureResponseDistance={
                {
                  "bottom": -1,
                  "end": -1,
                  "start": -1,
                  "top": -1,
                }
              }
              onGestureCancel={[Function]}
              sheetAllowedDetents="large"
              sheetCornerRadius={-1}
              sheetExpandsWhenScrolledToEdge={true}
              sheetGrabberVisible={false}
              sheetLargestUndimmedDetent="all"
              style={
                {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "zIndex": 0,
                }
              }
            >
              <View
                accessibilityElementsHidden={false}
                importantForAccessibility="auto"
                style={
                  [
                    {
                      "backgroundColor": "rgb(242, 242, 242)",
                      "flex": 1,
                    },
                    [
                      {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    {
                      "flex": 1,
                    }
                  }
                >
                  <View
                    style={
                      {
                        "margin": 5,
                      }
                    }
                  />
                </View>
                <View
                  onLayout={[Function]}
                  style={null}
                >
                  <View
                    collapsable={false}
                    pointerEvents="box-none"
                    style={
                      {
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                        "zIndex": 0,
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      style={
                        {
                          "backgroundColor": "rgb(255, 255, 255)",
                          "borderBottomColor": "rgb(216, 216, 216)",
                          "flex": 1,
                          "shadowColor": "rgb(216, 216, 216)",
                          "shadowOffset": {
                            "height": 0.5,
                            "width": 0,
                          },
                          "shadowOpacity": 0.85,
                          "shadowRadius": 0,
                        }
                      }
                    />
                  </View>
                  <View
                    collapsable={false}
                    pointerEvents="box-none"
                    style={
                      {
                        "height": 44,
                        "maxHeight": undefined,
                        "minHeight": undefined,
                        "opacity": undefined,
                        "transform": undefined,
                      }
                    }
                  >
                    <View
                      pointerEvents="none"
                      style={
                        {
                          "height": 0,
                        }
                      }
                    />
                    <View
                      pointerEvents="box-none"
                      style={
                        {
                          "alignItems": "stretch",
                          "flex": 1,
                          "flexDirection": "row",
                        }
                      }
                    >
                      <View
                        collapsable={false}
                        pointerEvents="box-none"
                        style={
                          {
                            "alignItems": "flex-start",
                            "flexBasis": 0,
                            "flexGrow": 1,
                            "justifyContent": "center",
                            "marginStart": 0,
                          }
                        }
                      >
                        <View
                          accessibilityRole="button"
                          accessibilityState={
                            {
                              "busy": undefined,
                              "checked": undefined,
                              "disabled": undefined,
                              "expanded": undefined,
                              "selected": undefined,
                            }
                          }
                          accessibilityValue={
                            {
                              "max": undefined,
                              "min": undefined,
                              "now": undefined,
                              "text": undefined,
                            }
                          }
                          accessible={true}
                          collapsable={false}
                          focusable={true}
                          onBlur={[Function]}
                          onClick={[Function]}
                          onFocus={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            {
                              "marginHorizontal": 11,
                              "opacity": 1,
                            }
                          }
                        >
                          <Image
                            fadeDuration={0}
                            source={
                              {
                                "testUri": "../../@react-navigation/drawer/lib/commonjs/views/assets/toggle-drawer-icon.png",
                              }
                            }
                            style={
                              [
                                {
                                  "height": 24,
                                  "margin": 3,
                                  "resizeMode": "contain",
                                  "width": 24,
                                },
                                null,
                              ]
                            }
                          />
                        </View>
                      </View>
                      <View
                        collapsable={false}
                        pointerEvents="box-none"
                        style={
                          {
                            "justifyContent": "center",
                            "marginHorizontal": 16,
                            "maxWidth": 590,
                          }
                        }
                      >
                        <Text
                          accessibilityRole="header"
                          aria-level="1"
                          collapsable={false}
                          numberOfLines={1}
                          style={
                            {
                              "color": "rgb(28, 28, 30)",
                              "fontSize": 17,
                              "fontWeight": "600",
                            }
                          }
                        >
                          Bets
                        </Text>
                      </View>
                      <View
                        collapsable={false}
                        pointerEvents="box-none"
                        style={
                          {
                            "alignItems": "flex-end",
                            "flexBasis": 0,
                            "flexGrow": 1,
                            "justifyContent": "center",
                            "marginEnd": 0,
                          }
                        }
                      />
                    </View>
                  </View>
                </View>
              </View>
            </RNSScreen>
          </RNSScreenNavigationContainer>
        </View>
        <View
          animatedProps={
            {
              "accessibilityElementsHidden": true,
              "importantForAccessibility": "no-hide-descendants",
              "pointerEvents": "none",
            }
          }
          style={
            [
              {
                "backgroundColor": "rgba(0, 0, 0, 0.5)",
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              {},
              {
                "opacity": undefined,
                "zIndex": -1,
              },
              {
                "backgroundColor": "rgba(0, 0, 0, 0.5)",
              },
            ]
          }
        >
          <View
            accessibilityLabel="Close drawer"
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "flex": 1,
                "pointerEvents": "auto",
              }
            }
          />
        </View>
      </View>
      <View
        removeClippedSubviews={false}
        style={
          [
            {
              "bottom": 0,
              "maxWidth": "100%",
              "top": 0,
              "width": "80%",
            },
            {
              "position": "absolute",
              "zIndex": 0,
            },
            {
              "transform": [
                {
                  "translateX": -320,
                },
              ],
            },
            [
              {
                "backgroundColor": "rgb(255, 255, 255)",
                "width": 320,
              },
              false,
              undefined,
            ],
          ]
        }
      >
        <View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "margin": 20,
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                {
                  "fontSize": 20,
                  "fontWeight": "bold",
                }
              }
            >
              X
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
